aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java

```diff
@@ -115,4 +118,25 @@ public String scheme() {
   public String toString() {
     return location;
   }
+
+  public S3URI toDirectoryPath() {
+    if (key.endsWith(PATH_DELIM)) {
+      return new S3URI(location);
+    }
+    return new S3URI(location + PATH_DELIM);
```

Comment:
should also inherit `bucketToAccessPointMapping` in constructor

Reply:
oh sorry I thought we preserve `bucketToAccessPointMapping` in URI, but looks like it is not. then nvm